qupath-gui-fx/src/main/java/qupath/lib/gui/charts/ScatterPlotChart.java

```diff
@@ -0,0 +1,337 @@
+package qupath.lib.gui.charts;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.chart.NumberAxis;
+import javafx.scene.chart.ScatterChart;
+import javafx.scene.chart.XYChart;
+import javafx.scene.input.MouseEvent;
+import qupath.lib.common.ColorTools;
+import qupath.lib.gui.tools.ColorToolsFX;
+import qupath.lib.gui.viewer.QuPathViewer;
+import qupath.lib.images.ImageData;
+import qupath.lib.objects.PathObject;
+import qupath.lib.objects.hierarchy.PathObjectHierarchy;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.function.Function;
+
+/**
+ * A scatter plot wrapping the JavaFX chart {@link ScatterChart} to add functionality like
+ */
+public class ScatterPlotChart extends ScatterChart<Number, Number> {
+    private final IntegerProperty rngSeed = new SimpleIntegerProperty(42);
```

Comment:
Please add a blank line here, for consistency with all our other classes & most Java code.

Most classes should also have a logger, since there is probably something, somewhere worth logging.